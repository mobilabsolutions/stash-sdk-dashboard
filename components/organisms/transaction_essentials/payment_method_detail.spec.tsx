import { testRender } from '../../../test_utils'
import PaymentMethodsDet from './payment_method_detail'

it('Payment Method should show PayPal details', () => {
  testRender(PaymentMethodsDet, {
    paymentMethod: 'PAY_PAL',
    personalData: {
      email: 'name@emailcom'
    }
  })
})

it('Payment Method should show Credit Cards details', () => {
  testRender(PaymentMethodsDet, {
    paymentMethod: 'CC',
    ccConfig: {
      ccExpiry: '12/25',
      ccHolderName: 'Alan Doe',
      ccMask: '3344',
      ccType: 'DINERS'
    }
  })
})

it('Payment Method should show SEPA details', () => {
  testRender(PaymentMethodsDet, {
    paymentMethod: 'SEPA',
    sepaConfig: {
      bic: '87f387hf383',
      iban: '7551210800124512619'
    }
  })
})
