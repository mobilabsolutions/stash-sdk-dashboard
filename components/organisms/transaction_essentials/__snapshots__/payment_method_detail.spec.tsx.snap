// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Method should show Credit Cards details 1`] = `
<div>
  <div
    className="payment_method_detail__FLeftIcon-sc-1x0zhjy-1 jsLaPA"
  >
    <div
      title="DINERS"
    >
      <svg
        height={20}
        style={Object {}}
        viewBox="0 0 22 18"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="a-dinners"
            x1="70.752%"
            x2="32.476%"
            y1="81.094%"
            y2="18.906%"
          >
            <stop
              offset="0%"
              stopColor="#009BE0"
            />
            <stop
              offset="100%"
              stopColor="#006BA8"
            />
          </linearGradient>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M12.765 17.183c4.718.023 9.025-3.83 9.025-8.517 0-5.125-4.307-8.668-9.025-8.666h-4.06C3.93-.002 0 3.542 0 8.666c0 4.688 3.93 8.54 8.705 8.517h4.06z"
            fill="url(#a-dinners)"
          />
          <path
            d="M8.721.716C4.358.717.823 4.238.822 8.582c.001 4.344 3.536 7.864 7.9 7.865 4.363-.001 7.9-3.521 7.9-7.865S13.085.717 8.722.716zM3.714 8.582A4.994 4.994 0 0 1 6.93 3.93v9.305a4.992 4.992 0 0 1-3.215-4.652zm6.8 4.654V3.929a4.992 4.992 0 0 1 3.216 4.653 4.993 4.993 0 0 1-3.217 4.654z"
            fill="#F4F6F7"
          />
        </g>
      </svg>
    </div>
  </div>
  <span
    className="payment_method_detail__FLeftText-sc-1x0zhjy-0 gTezeC"
  >
    <span
      className="payment_method_detail__CCDATA-sc-1x0zhjy-2 fXChlu"
    >
      3344
    </span>
    <span
      className="payment_method_detail__CCDATA-sc-1x0zhjy-2 fXChlu"
    >
      12/25
    </span>
  </span>
</div>
`;

exports[`Payment Method should show PayPal details 1`] = `
<div>
  <div
    className="payment_method_detail__FLeftIcon-sc-1x0zhjy-1 jsLaPA"
  >
    <svg
      height={20}
      viewBox="0 0 60 40"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="nonzero"
      >
        <path
          d="M22.616 38.583l.692-4.397-1.542-.035h-7.362L19.52 1.71a.418.418 0 0 1 .416-.355h12.413c4.122 0 6.966.858 8.45 2.55.697.794 1.14 1.624 1.354 2.537.225.958.23 2.103.01 3.5l-.016.101v.895l.696.394c.531.27 1.009.634 1.41 1.075.595.679.98 1.542 1.143 2.565.168 1.052.112 2.304-.163 3.721-.318 1.63-.831 3.05-1.525 4.213a8.664 8.664 0 0 1-2.415 2.646c-.921.654-2.016 1.15-3.253 1.468-1.199.312-2.566.47-4.065.47h-.967a2.93 2.93 0 0 0-1.888.695 2.925 2.925 0 0 0-.985 1.757l-.072.396-1.223 7.748-.056.285c-.014.09-.04.135-.077.165a.205.205 0 0 1-.127.047h-5.964z"
          fill="#253B80"
        />
        <path
          d="M43.502 10.147c-.037.236-.08.479-.127.727-1.637 8.405-7.238 11.309-14.391 11.309h-3.642c-.875 0-1.612.635-1.748 1.498l-1.865 11.826-.528 3.352a.932.932 0 0 0 .92 1.077h6.46c.764 0 1.414-.555 1.535-1.31l.063-.328 1.216-7.718.078-.424c.12-.756.77-1.312 1.535-1.312h.967c6.258 0 11.157-2.541 12.59-9.894.597-3.072.288-5.637-1.295-7.44a6.176 6.176 0 0 0-1.768-1.363z"
          fill="#179BD7"
        />
        <path
          d="M41.79 9.464a12.912 12.912 0 0 0-1.593-.354 20.227 20.227 0 0 0-3.21-.234h-9.73c-.765 0-1.416.557-1.534 1.313l-2.07 13.11-.06.382a1.768 1.768 0 0 1 1.749-1.498h3.642c7.153 0 12.754-2.905 14.39-11.309.05-.248.09-.49.128-.727a8.727 8.727 0 0 0-1.346-.568 11.97 11.97 0 0 0-.367-.115z"
          fill="#222D65"
        />
        <path
          d="M25.723 10.189a1.547 1.547 0 0 1 1.534-1.312h9.73c1.152 0 2.228.076 3.21.235.665.104 1.32.26 1.96.467.483.16.932.35 1.346.568.487-3.106-.004-5.221-1.683-7.136C39.968.903 36.627 0 32.35 0H19.938c-.873 0-1.618.635-1.753 1.5l-5.17 32.774a1.067 1.067 0 0 0 1.051 1.233h7.664l1.924-12.208 2.07-13.11z"
          fill="#253B80"
        />
      </g>
    </svg>
  </div>
  <span
    className="payment_method_detail__FLeftText-sc-1x0zhjy-0 gTezeC"
  >
    name@emailcom
  </span>
</div>
`;

exports[`Payment Method should show SEPA details 1`] = `
<div>
  <div
    className="payment_method_detail__FLeftIcon-sc-1x0zhjy-1 jsLaPA"
  >
    <svg
      height={20}
      viewBox="0 0 60 40"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M13.314 16.227h-4.1c0-.674-.08-1.125-.237-1.355-.244-.34-.917-.511-2.02-.511-1.07 0-1.78.098-2.126.294-.346.197-.52.625-.52 1.284 0 .596.154.99.462 1.177.22.134.511.212.874.236l.827.058c1.772.118 2.879.2 3.32.248 1.402.141 2.418.514 3.048 1.119.496.471.792 1.087.887 1.848.055.455.082.97.082 1.543 0 1.319-.125 2.285-.377 2.897-.457 1.123-1.482 1.833-3.078 2.13-.669.127-1.692.19-3.067.19-2.296 0-3.897-.137-4.8-.412-1.11-.337-1.84-1.019-2.194-2.046C.098 24.355 0 23.403 0 22.069h4.1v.341c0 .712.205 1.17.613 1.373.283.14.624.214 1.025.223h1.508c.769 0 1.26-.04 1.472-.119.377-.148.624-.387.742-.717.063-.203.094-.466.094-.788 0-.722-.263-1.16-.789-1.318-.196-.062-1.11-.149-2.742-.259-1.31-.094-2.221-.184-2.73-.27-1.343-.252-2.238-.769-2.684-1.553-.393-.667-.59-1.675-.59-3.024 0-1.027.107-1.85.319-2.47.212-.62.554-1.09 1.026-1.413.691-.494 1.576-.784 2.652-.87.896-.078 1.89-.118 2.983-.118 1.72 0 2.947.099 3.678.295 1.785.48 2.677 1.823 2.677 4.03 0 .182-.014.453-.04.815M35.752 19.354H38.5c.871-.032 1.401-.099 1.592-.204.26-.143.435-.43.522-.86.056-.287.084-.712.084-1.278 0-.693-.056-1.207-.167-1.542-.158-.47-.538-.756-1.14-.86a8.909 8.909 0 0 0-.854-.024h-2.785v4.768zm-4.407 8.03V11h8.211c1.127 0 1.986.092 2.577.276 1.35.423 2.256 1.29 2.72 2.6.24.688.36 1.723.36 3.105 0 1.662-.133 2.857-.396 3.584-.526 1.438-1.61 2.266-3.248 2.48-.192.033-1.011.062-2.457.085l-.732.024h-2.628v4.23h-4.407zM53.746 21.361l-2.065-7.102-2.004 7.102h4.069zm.875 3.192h-5.893l-.791 2.832h-4.56L48.297 11h6.648l4.981 16.385h-4.464l-.84-2.832z"
          fill="#0B4A8E"
        />
        <path
          d="M25.262 13.305c1.818 0 3.44.838 4.507 2.15l.95-2.038a8.326 8.326 0 0 0-5.74-2.271c-3.472 0-6.443 2.098-7.662 5.069h-1.853L14.37 18.56h2.384c-.018.218-.03.436-.03.658 0 .269.014.535.04.797h-1.197l-1.094 2.347h2.9c1.255 2.896 4.188 4.929 7.607 4.929a8.349 8.349 0 0 0 4.645-1.399v-2.877a5.79 5.79 0 0 1-4.363 1.982 5.804 5.804 0 0 1-4.86-2.635h6.074l1.094-2.347h-8.061a5.943 5.943 0 0 1-.035-1.455h8.774l1.093-2.346h-9.108a5.807 5.807 0 0 1 5.029-2.91"
          fill="#F6BC25"
        />
      </g>
    </svg>
  </div>
  <span
    className="payment_method_detail__FLeftText-sc-1x0zhjy-0 gTezeC"
  >
    <span>
      7551 2108 0012 45126 19
    </span>
  </span>
</div>
`;
